modules/040-node-manager/hooks/handle_draining.go

```diff
@@ -189,6 +208,48 @@ func handleDraining(input *go_hook.HookInput, dc dependency.Container) error {
 	return nil
 }
 
+func getDrainTimeout(input *go_hook.HookInput, client k8s.Client, ngName string) time.Duration {
+	nodeGroups, err := client.Dynamic().Resource(nodeGroupResource).Namespace("").List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		input.Logger.Error("Failed to list node groups")
+		return defaultDrainTimeout
+	}
+
+	for _, group := range nodeGroups.Items {
+		groupName := group.GetName()
+		if groupName != ngName {
+			continue
+		}
+
+		spec, found, err := unstructured.NestedMap(group.Object, "spec")
```

Comment:
second, better way marshal it to nedegroup struct
modules/040-node-manager/hooks/internal/v1/nodegroup.go

Reply:
we do not need in this case multiple is's with checking types and existstings fields

Reply:
changed to ngv1.NodeGroup